cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -562,7 +567,63 @@ func (c *CStorPoolController) syncCsp(cStorPool *apis.CStorPool) {
 		c.recorder.Event(cStorPool, corev1.EventTypeWarning, string(common.FailureCapacitySync), string(common.MessageResourceFailCapacitySync))
 	} else {
 		cStorPool.Status.Capacity = *capacity
+
+		qn, err := convertToBytes([]string{capacity.Total, capacity.Free, capacity.Used})
+		if err != nil {
+			klog.Errorf("Failed to parse capacity.. err=%s", err)
+			return
+		}
+
+		total, _, used := qn[0], qn[1], qn[2]
+		usedCapacity := (used * 100) / total
+		if (int(usedCapacity) >= cStorPool.Spec.PoolSpec.ROThresholdLimit) &&
+			(cStorPool.Spec.PoolSpec.ROThresholdLimit != 0 &&
+				cStorPool.Spec.PoolSpec.ROThresholdLimit != 100) {
+			if !cStorPool.Status.ReadOnly {
+				if err = pool.SetPoolRDMode(cStorPool, true); err != nil {
```

Comment:
did we covered the cases where pool.SetPoolRDMode is done, but, update of cStorPool.Status.ReadOnly as true in etcd is not done due to any reason like restart of pool-mgmt / update to etcd failed?